contents/docs/error-tracking/installation/nextjs.mdx

```diff
@@ -0,0 +1,179 @@
+      }
+    )
+  }
+
+  return posthogInstance
+}
+```
+
+You can now use the `getPostHogServer` function to capture exceptions in server-side code.
+
+```js
+const posthog = getPostHogServer()
+
+posthog.capture("button-clicked")
+```
+
+</Step>
+
+<Step checkpoint title="Verify server-side exceptions" subtitle="Confirm capture is working server-side">
+
+You should see events and exceptions in PostHog coming from your server-side code in the activity feed.
+
+<div className="mt-4 mb-2">
+  <CallToAction type="primary" to="https://app.posthog.com/activity/explore">
+    Check for server events in PostHog
+  </CallToAction>
+</div>
+
+</Step>
+
+<Step title="Capturing server-side exceptions" subtitle="Your goal in this step: Capture your first server-side exception event." badge="required">
+
+To capture errors that occur in your server-side code, you can set up a [`instrumentation.ts`](https://nextjs.org/docs/app/building-your-application/optimizing/instrumentation) file at the root of your project. This provides a `onRequestError` hook that you can use to capture errors.
+
+Importantly, you need to:
+
+1. Set up a `posthog-node` client in your server-side code. See our doc on [setting up Next.js server-side analytics](/docs/libraries/next-js#server-side-analytics) for more.
+2. Check the request is running in the `nodejs` runtime to ensure PostHog works. You can call `posthog.debug()` to get verbose logging.
+3. Get the `distinct_id` from the cookie to connect the error to a specific user.
+
+This looks like this:
+
+```js
+// instrumentation.js
+export function register() {
+  // No-op for initialization
+}
+
+export const onRequestError = async (err, request, context) => {
+  if (process.env.NEXT_RUNTIME === 'nodejs') {
+    const { getPostHogServer } = require('./lib/posthog-server')
+    const posthog = await getPostHogServer()
+    let distinctId = null
+    if (request.headers.cookie) {
+      const cookieString = request.headers.cookie
+      const postHogCookieMatch = cookieString.match(/ph_phc_.*?_posthog=([^;]+)/)
+
+      if (postHogCookieMatch && postHogCookieMatch[1]) {
+        try {
+          const decodedCookie = decodeURIComponent(postHogCookieMatch[1])
+          const postHogData = JSON.parse(decodedCookie)
+          distinctId = postHogData.distinct_id
+        } catch (e) {
+          console.error('Error parsing PostHog cookie:', e)
+        }
+      }
+    }
+
+    await posthog.captureException(err, distinctId || undefined)
+  }
+}
+```
+
+You can find a full example of both this and client-side error tracking in our [Next.js error monitoring tutorial](/tutorials/nextjs-error-monitoring).
+
+</Step>
+
+<Step title="Next steps">
+
+After configuring PostHog, you will start receiving exceptions in PostHog. To help you get more out of your error tracking, you can:
+
+1. [Upload source maps](/docs/error-tracking/upload-source-maps/nextjs) – Required for accurate stack traces if you serve minified bundles.
```

Comment:
Getting customers to upload source maps is certainly the most difficult step of our onboarding. It kind of feels like when I get to the end of the installation steps that I'm done. It's not clear to me that this next step is very much also required. We handled this in the old docs by having the source map steps duplicated at the bottom of the installation flow.

I think we should have another `<Step .../>` here that links you to the source map upload process for your framework e.g. once you've verified you're getting events then click here to upload source maps